packages/commands/src/solana/depositAndCall.ts

```diff
@@ -0,0 +1,179 @@
+import * as anchor from "@coral-xyz/anchor";
+import { Wallet } from "@coral-xyz/anchor";
+import {
+  ASSOCIATED_TOKEN_PROGRAM_ID,
+  TOKEN_PROGRAM_ID,
+} from "@solana/spl-token";
+import { PublicKey } from "@solana/web3.js";
+import GATEWAY_DEV_IDL from "@zetachain/protocol-contracts-solana/dev/idl/gateway.json";
+import GATEWAY_PROD_IDL from "@zetachain/protocol-contracts-solana/prod/idl/gateway.json";
+import { ethers } from "ethers";
+import { z } from "zod";
+
+import { SOLANA_TOKEN_PROGRAM } from "../../../../types/shared.constants";
+import { handleError, validateAndParseSchema } from "../../../../utils";
+import { parseAbiValues } from "../../../../utils/parseAbiValues";
+import {
+  confirmSolanaTx,
+  createSolanaCommandWithCommonOptions,
+  getAPI,
+  getKeypair,
+  getSPLToken,
+  isSOLBalanceSufficient,
+  solanaDepositAndCallOptionsSchema,
+} from "../../../../utils/solana.commands.helpers";
+
+type DepositAndCallOptions = z.infer<typeof solanaDepositAndCallOptionsSchema>;
+
+const main = async (options: DepositAndCallOptions) => {
+  // Mainnet and devnet use the same IDL
+  const gatewayIDL =
+    options.network === "localnet" ? GATEWAY_DEV_IDL : GATEWAY_PROD_IDL;
+
+  const keypair = await getKeypair({
+    mnemonic: options.mnemonic,
+    name: options.name,
+    privateKey: options.privateKey,
+  });
+
+  const API = getAPI(options.network);
+
+  const connection = new anchor.web3.Connection(API);
+  const provider = new anchor.AnchorProvider(connection, new Wallet(keypair));
+  const gatewayProgram = new anchor.Program(gatewayIDL as anchor.Idl, provider);
+
+  const receiverBytes = ethers.getBytes(options.recipient);
+  const stringifiedTypes = JSON.stringify(options.types);
+  const values = parseAbiValues(stringifiedTypes, options.values);
+  const abiCoder = ethers.AbiCoder.defaultAbiCoder();
+  const encodedParameters = abiCoder.encode(options.types, values);
+  const message = Buffer.from(encodedParameters.slice(2), "hex");
+
+  const revertOptions = {
+    abortAddress: ethers.getBytes(options.abortAddress),
+    callOnRevert: options.callOnRevert,
+    onRevertGasLimit: new anchor.BN(options.onRevertGasLimit ?? 0),
+    revertAddress: options.revertAddress
+      ? new PublicKey(options.revertAddress)
+      : provider.wallet.publicKey,
+    revertMessage: Buffer.from(options.revertMessage, "utf8"),
+  };
```

Comment:
This `revertOptions` object seems to be the same across all 3 files. We should consider extracting into `solana.commands.helpers.ts`